### ISSUE TITLE
GPS disconnects when I make a call

### ISSUE THREAD

**Issue by gusioo** [BasicAirData/GPSLogger#4]
Hello,

Some time ago I open a new issue #3 with GPS that stops tracking in specific conditions. Now I know what is "specific conditions".

GPS stop tracking when I make a call or answering a call. For example:

I start tracking with GPS Logger.
I use home button to minimize the app.
I make a call.

And now GPS is disconnected and GPS Logger stops the recording.

But when I enable "airplane mode" (sometimes called "offline mode") in my phone GPS Logger works very well.

Thanks for help.
Jacek.

**Comment by GrazianoCapelli** [comment]
Hello.
I've checked your specific condition and sometimes I faced this problem also with my phone.
But the problem occurs only if I use the Back button to minimize the app (not with the Home button).
Today we released a new update. We have tested it with many phones, the problem seems gone.
Please try it and, in case, report it again.

In any case we are evaluating to move the recording process from the application to a background service, that has a more "solid" life cycle.

Thanks for your useful reporting!

**Comment by GrazianoCapelli** [comment]
Sadly the reported problem still present.
We'll move the GPS location service and the logging process to a background service in the next version.
We keep for now the issue open.
Thanks again.

**Comment by gusioo** [comment]
> Sadly the reported problem still present.

Unfortunately yes - today I went to walk with my kids and I used GPS Logger. After that my wife called me - GPS was turned off.

But I think that next update resolve this problem - your app is fantastic.

**Comment by GrazianoCapelli** [comment]
In version 2.1.1 we added a foreground service in order to keep up the application also in long running background records. While the GPSLogger icon is present in the notification area, the app is active.
We checked successfully the background stability with many hardware and Android versions: no more unwanted stops, recording still active now and rocks solid.

Now the smartphone buttons behavior is the following:
- HOME button brings the app in background;
- BACK button stops the service and closes the app (with confirmation dialog if needed);

I hope you will test it and, in case, report again.
In any case thank you very much for your cooperation and happy tracking!
